ptrace/debugger/syscall_state.py

```diff
@@ -37,7 +38,10 @@ def exit(self):
                 and not self.process.debugger.trace_exec:
             # Ignore the SIGTRAP after exec() syscall exit
             self.process.syscall()
-            self.process.waitSignals(SIGTRAP, SIGTRAP | 0x80)
+            signum = SIGTRAP
+            if self.process.debugger.use_sysgood:
+                signum |= SYSGOOD_FLAG
```

Comment:
ah, that sounds better! it's correct according to https://man7.org/linux/man-pages/man2/ptrace.2.html